mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/DeleteByQueryPerformer.java

```diff
@@ -41,20 +41,26 @@
     private final ClientProvider clientProvider;
     private final ExecutorService executor;
     private final int batchSize;
+    private final boolean isSynchronous;
 
     @Inject
     public DeleteByQueryPerformer(ClientProvider clientProvider, @Named("AsyncExecutor") ExecutorService executor) {
-        this(clientProvider, executor, DEFAULT_BATCH_SIZE);
+        this(clientProvider, executor, DEFAULT_BATCH_SIZE, false);
     }
 
-    public DeleteByQueryPerformer(ClientProvider clientProvider, @Named("AsyncExecutor") ExecutorService executor, int batchSize) {
+    public DeleteByQueryPerformer(ClientProvider clientProvider, @Named("AsyncExecutor") ExecutorService executor, int batchSize, boolean isSynchronous) {
         this.clientProvider = clientProvider;
         this.executor = executor;
         this.batchSize = batchSize;
+        this.isSynchronous = isSynchronous;
     }
 
     public void perform(QueryBuilder queryBuilder) {
```

Comment:
Doesn't it make sense to override a method in tests than to introduce some code used only for testing in your implementation?